### ISSUE TITLE
Single round trip updates

### ISSUE THREAD

**Issue by YohDeadfall** [dotnet/efcore#19631]
@ajcvickers recently posted [a blog post](https://blog.oneunicorn.com/2020/01/17/dontcallupdate/) where it recommends to use two round trip updates. Here is an proposal which combines two ways and which should give some perf improvements.

```csharp
var userExists = await context.Users
    .Update(user => user.Email = email)
    .Where(user => user.Name == name)
    .AnyAsync();
```

The `Update` method should generate an `UPDATE` query to a database. Other methods should behave as before, maybe not all but mostly all. Therefore, the resulting query should look like this:

```sql
UPDATE user SET email = @email WHERE name = @name;
```

This query can be executed by SQL Server and PostgreSQL and returns number of affected rows. The number can be taken from the `ExecuteNonQueryAsync` method of a data reader. If result greater than zero, than all things done.

In case when an user wants to get updated entities back, there is the `RETURNING` clause in PostgreSQL and `OUTPUT` in SQL Server:

```csharp
var users = await context.Users
    .Update(user => user.Email = email)
    .Where(user => user.Name == name)
    .AsAsyncEnumerable();
```
```sql
-- PostgreSQL
UPDATE user SET email = @email WHERE name = @name RETURNING name, email;

-- SQL Server
UPDATE user SET email = @email WHERE name = @name OUTPUT name, email;
```

Need something to do with users before returning? Easily using PostgreSQL and CTEs. In case of SQL Server this could be tricky and require a temporary table.

While it won't give EF Core the same functionality as raw SQL, the feature could make it a bit powerful and extend number of scenarios where it can be used.

**Comment by ajcvickers** [comment]
@YohDeadfall This looks like a duplicate of #795, unless I am missing something.

**Comment by YohDeadfall** [comment]
Mostly looks like it is, but with a small difference. This one isn't limited by no-loading, moreover it allows it when requested as shown in the last example.

Anyway, this issue can be closed as duplicate if #795 gets changed a bit (at least the title).
